tests/core/test_single_derived_signal.py

```diff
@@ -0,0 +1,161 @@
+import asyncio
+from unittest.mock import call
+
+import pytest
+
+from ophyd_async.core import (
+    AsyncStatus,
+    Device,
+    DeviceVector,
+    StandardReadable,
+    StrictEnum,
+    derived_signal_r,
+    derived_signal_rw,
+    derived_signal_w,
+    soft_signal_rw,
+)
+from ophyd_async.testing import (
+    assert_describe_signal,
+    assert_reading,
+    assert_value,
+    get_mock,
+)
+
+
+class BeamstopPosition(StrictEnum):
+    IN_POSITION = "In position"
+    OUT_OF_POSITION = "Out of position"
+
+
+class Beamstop(Device):
+    def __init__(self, name=""):
+        # Raw signals
+        self.x = soft_signal_rw(float)
+        self.y = soft_signal_rw(float)
+        # Derived signals
+        self.position = derived_signal_r(self._get_position, x=self.x, y=self.y)
+        super().__init__(name=name)
+
+    def _get_position(self, x: float, y: float) -> BeamstopPosition:
+        if abs(x) < 1 and abs(y) < 2:
+            return BeamstopPosition.IN_POSITION
+        else:
+            return BeamstopPosition.OUT_OF_POSITION
+
+
+class Positioner(Device):
+    def __init__(self, name=""):
+        # Raw signals
+        self.x = soft_signal_rw(float)
+        self.y = soft_signal_rw(float)
+        # Derived signals
+        self.position = derived_signal_rw(
+            self._get_position, self.set, x=self.x, y=self.y
+        )
+        super().__init__(name=name)
+
+    def _get_position(self, x: float, y: float) -> BeamstopPosition:
+        if abs(x) < 1 and abs(y) < 2:
+            return BeamstopPosition.IN_POSITION
+        else:
+            return BeamstopPosition.OUT_OF_POSITION
+
+    @AsyncStatus.wrap
+    async def set(self, value: BeamstopPosition) -> None:
```

Comment:
Should: Naming this set is ambiguous as it means I can now do either `yield from bps.abs_set(device, IN)` or  `yield from bps.abs_set(device.position, IN)`? I would rename this to something like `_set_from_position` to be clear that the device is not designed to be used like `yield from bps.abs_set(device, IN)`.

Reply:
Can you think of a usecase where we would want the `set` method publicly accessible on the device? Or is this never the case for a single derived signal?